docs/_scripts/gen_api_ref_pages.py

```diff
@@ -0,0 +1,32 @@
+"""Generate API reference pages."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("src").rglob("*.py")):
+    module_path = path.relative_to("src").with_suffix("")
+    doc_path = path.relative_to("src").with_suffix(".md")
+    full_doc_path = Path("reference", doc_path)
+
+    parts = list(module_path.parts)
```

Comment:
If I understand correctly, you are using `parts` read-only below, so it can stay as a tuple, but your call.

Reply:
This code comes verbatim from the [recipe in the `mkdocstrings` docs](https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages), since it works, I didn't further refine it 😅